cli/cmd/install_k0s.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Codesphere Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"fmt"
+
+	packageio "github.com/codesphere-cloud/cs-go/pkg/io"
+	"github.com/spf13/cobra"
+
+	"github.com/codesphere-cloud/oms/internal/env"
+	"github.com/codesphere-cloud/oms/internal/installer"
+	"github.com/codesphere-cloud/oms/internal/portal"
+	"github.com/codesphere-cloud/oms/internal/util"
+)
+
+// InstallK0sCmd represents the k0s download command
+type InstallK0sCmd struct {
+	cmd        *cobra.Command
+	Opts       InstallK0sOpts
+	Env        env.Env
+	FileWriter util.FileIO
+}
+
+type InstallK0sOpts struct {
+	*GlobalOptions
+	Config string
+	Force  bool
+}
+
+func (c *InstallK0sCmd) RunE(_ *cobra.Command, args []string) error {
+	hw := portal.NewHttpWrapper()
+	env := c.Env
+	k0s := installer.NewK0s(hw, env, c.FileWriter)
+
+	if !k0s.BinaryExists() || c.Opts.Force {
+		err := k0s.Download(c.Opts.Force, false)
+		if err != nil {
+			return fmt.Errorf("failed to download k0s: %w", err)
+		}
+	}
+
+	err := k0s.Install(c.Opts.Config, c.Opts.Force)
+	if err != nil {
+		return fmt.Errorf("failed to install k0s: %w", err)
+	}
+
+	return nil
+}
+
+func AddInstallK0sCmd(install *cobra.Command, opts *GlobalOptions) {
+	k0s := InstallK0sCmd{
+		cmd: &cobra.Command{
+			Use:   "k0s",
+			Short: "Install k0s Kubernetes distribution",
+			Long: packageio.Long(`Install k0s, a zero friction Kubernetes distribution, 
+				using a Go-native implementation. This will download the k0s 
```

Comment:
what does Go-native mean? Doesn't the install command just execute the k0s binary?